### PR TITLE
DC-2098: Play 2.6 upgrade

### PR DESCRIPTION
**Pull request by rwalpole**

### CODE REVIEW

File: app/uk/gov/hmrc/ups/connectors/PreferencesConnector.scala [hunk 1]
```diff
@@ -16,24 +16,30 @@
...
-trait PreferencesConnector {
-  implicit val optionalPullItemReads = new HttpReads[Option[PulledItem]] {
-    override def read(method: String, url: String, response: HttpResponse) =
+import scala.concurrent.duration._
+
+@Singleton
+class PreferencesConnector @Inject()(httpClient: HttpClient, runMode: RunMode, configuration: Configuration)
+                                    (implicit ec: ExecutionContext) {
+
+  val servicesConfig = new ServicesConfig(configuration,runMode)
```
Comment (muralishub) [inline app/uk/gov/hmrc/ups/connectors/PreferencesConnector.scala]: Can we inject ServicesConfig ?

File: app/uk/gov/hmrc/ups/connectors/PreferencesConnector.scala [hunk 2]
```diff
@@ -43,48 +49,35 @@ trait PreferencesConnector {
...
 
-  def changeStatus(callbackUrl: String, status: ProcessingStatus)(implicit hc: HeaderCarrier) =
-    http.POST[JsValue, Int](s"$serviceUrl$callbackUrl", Json.obj("status" -> status.name))
-
-  def retryFailedUpdatesAfter: Duration
+  def changeStatus(callbackUrl: String, status: ProcessingStatus)(implicit hc: HeaderCarrier): Future[Int] =
+    httpClient.POST[JsValue, Int](s"$serviceUrl$callbackUrl", Json.obj("status" -> status.name))
 
-  def dateTimeFor(duration: Duration): DateTime = DateTimeUtils.now.minus(duration)
+  def dateTimeFor(duration: Duration): DateTime = DateTimeUtils.now.minus(duration.toMillis)
```
Comment (muralishub) [inline app/uk/gov/hmrc/ups/connectors/PreferencesConnector.scala:72]: Do we need to specify implicits for joda reads and writes? Not sure if this is relevant here

File: build.sbt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      ExternalService("USER_DETAILS")
+  ))
+  .configs(IntegrationTest)
+  .settings(integrationTestSettings(): _*)
+  .settings(inConfig(IntegrationTest)(Defaults.itSettings): _*)
+  .settings(
+    Keys.fork in IntegrationTest := false,
+    unmanagedSourceDirectories in IntegrationTest := (baseDirectory in IntegrationTest) (base => Seq(base / "it")).value,
+    addTestReportOption(IntegrationTest, "int-test-reports"),
+    //testGrouping in IntegrationTest := oneForkedJvmPerTest((definedTests in IntegrationTest).value),
```
Comment (muralishub) [inline build.sbt]: We can remove this comment

File: it/uk/gov/hmrc/ups/scheduler/CollectionRemovalISpec.scala [hunk 4]
```diff
@@ -1,37 +1,64 @@
...
     s"Test.removeOlderCollections.durationInDays" -> expirationPeriod,
     s"Test.scheduling.removeOlderCollections.initialDelay" -> "10 days",
-    s"Test.scheduling.removeOlderCollections.interval" -> "24 hours"
-  ))
+    s"Test.scheduling.removeOlderCollections.interval" -> "24 hours").build()
+
+//  override def additionalConfig: Map[String, Any] = Map(
+//    s"mongodb.uri" -> s"mongodb://localhost:27017/$databaseName",
+//    s"Test.removeOlderCollections.durationInDays" -> expirationPeriod,
+//    s"Test.scheduling.removeOlderCollections.initialDelay" -> "10 days",
```
Comment (muralishub) [inline it/uk/gov/hmrc/ups/scheduler/CollectionRemovalISpec.scala]: We can remove this comment

File: project/AppDependencies.scala [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    "uk.gov.hmrc"             %% "play-scheduling"        % "7.1.0-play-26",
+    "uk.gov.hmrc"             %% "domain"                 % "5.6.0-play-26",
+    "uk.gov.hmrc"             %% "work-item-repo"         % "6.8.0-play-26"
+  )
+
+  val test = Set(
+    "uk.gov.hmrc"             %% "reactivemongo-test"       % "4.15.0-play-26"    % "test,it",
+    "org.mockito"             %  "mockito-all"              % "1.9.5"             % "test,it",
+    "com.github.tomakehurst"  % "wiremock-jre8"             % "2.25.1"            % "test,it",
+    "org.scalatest"           %% "scalatest"                % "3.0.8"             % "test,it",
```
Comment (muralishub) [inline project/AppDependencies.scala:37]: Do we need to include scala test?

File: test/uk/gov/hmrc/ups/connectors/EntityResolverConnectorSpec.scala [hunk 6]
```diff
@@ -18,62 +18,78 @@ package uk.gov.hmrc.ups.connectors
...
+
   }
 
 }
 
-class HttpEntityResolverConnector extends EntityResolverConnector with MockHttpGet {
-  def serviceUrl = "baseUrl"
-}
+//class HttpEntityResolverConnector extends EntityResolverConnector with MockHttpGet {
+//  def serviceUrl = "baseUrl"
```
Comment (muralishub) [inline test/uk/gov/hmrc/ups/connectors/EntityResolverConnectorSpec.scala]: We can remove this comment

File: test/uk/gov/hmrc/ups/repository/RandomDataGenerator.scala [hunk 7]
```diff
@@ -18,24 +18,33 @@ package uk.gov.hmrc.ups.repository
...
+  implicit val ec: ExecutionContextExecutor = scala.concurrent.ExecutionContext.Implicits.global
 
   val BATCH_SIZE: Int = 100000
 
   "RandomDataGenerator" should {
 
     "create 3M random records in one day" in new TestSetup {
-
-      val repository: UpdatedPrintSuppressionsRepository = new UpdatedPrintSuppressionsRepository(new LocalDate().minusDays(1), MongoCounterRepository())
+      //implcit val mongoDb = mongo
```
Comment (muralishub) [inline test/uk/gov/hmrc/ups/repository/RandomDataGenerator.scala]: We can remove this comment

Comment (rwalpole) [reply test/uk/gov/hmrc/ups/repository/RandomDataGenerator.scala]: Removed.

File: test/uk/gov/hmrc/ups/connectors/EntityResolverConnectorSpec.scala (hunk 6 above)
Comment (rwalpole) [reply test/uk/gov/hmrc/ups/connectors/EntityResolverConnectorSpec.scala]: Removed.

File: project/AppDependencies.scala (hunk 5 above)
Comment (rwalpole) [reply project/AppDependencies.scala:37]: I tried removing it but it seems we do need it.

File: it/uk/gov/hmrc/ups/scheduler/CollectionRemovalISpec.scala (hunk 4 above)
Comment (rwalpole) [reply it/uk/gov/hmrc/ups/scheduler/CollectionRemovalISpec.scala]: Removed.

File: build.sbt (hunk 3 above)
Comment (rwalpole) [reply build.sbt]: Removed.

File: app/uk/gov/hmrc/ups/connectors/PreferencesConnector.scala (hunk 1 above)
Comment (rwalpole) [reply app/uk/gov/hmrc/ups/connectors/PreferencesConnector.scala]: Yes we can.

File: app/uk/gov/hmrc/ups/connectors/PreferencesConnector.scala (hunk 2 above)
Comment (rwalpole) [reply app/uk/gov/hmrc/ups/connectors/PreferencesConnector.scala:72]: Updated.

**Review by lxol — COMMENTED** [review]
one unit test fails locally
```
info] UpdatedPrintSuppressionsDatabaseSpec:
[info] collections list repo                                                                                                                                                 [info] - should return a list of UPS collections *** FAILED ***
[info]   List("updated_print_suppressions_20191129", "updated_print_suppressions_20190512", "updated_print_suppressions_20190513", "updated_print_suppressions_20191128") did
 not contain only ("updated_print_suppressions_20191129", "updated_print_suppressions_20191128") (UpdatedPrintSuppressionsDatabaseSpec.scala:67)
[info] - drop collection and return true if successful
```

**Pull request merged**